src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -793,7 +793,13 @@ public static String getType(Field field, String propertyName,
     public static String getDefaultValue(String propertyName, Field field,
                                          Object classInstance, String moduleName)
             throws MacroExecutionException {
-        final Object value = getFieldValue(field, classInstance);
+        final Object value;
+        if (field != null) {
+            value = getFieldValue(field, classInstance);
```

Comment:
Can we move this comparison to null for field in some upper by call stack method?
It sounds weird that we know that field is null but still asking getDefaultValue.
We should not call this method if field is null

Reply:
Such check for null should be at top of https://github.com/checkstyle/checkstyle/blob/9e03417b818f8e53ae1212453b90b884f0fb02a4/src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java#L465

Reply:
Existing code is good